### PR TITLE
bugfix: CUDA acceleration not working

### PR DESCRIPTION
**Pull request by fakezeta**
CUDA not working after #2286.
Refactored the code to be more polish

**Description**

This PR fixes missing libraries and error in CUDA environment.

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [X] Yes, I signed my commits.

### CODE REVIEW

File: backend/python/transformers/requirements.txt [hunk 1]
```diff
@@ -3,4 +3,7 @@ transformers
 grpcio==1.64.0
 protobuf
 torch
-certifi
\ No newline at end of file
+certifi
+intel-extension-for-transformers
```
Comment (mudler) [inline backend/python/transformers/requirements.txt:7]: looks something should go in https://github.com/mudler/LocalAI/blob/master/backend/python/transformers/requirements-intel.txt ? or is this for some other reason?

File: backend/python/transformers/backend.py [hunk 2]
```diff
@@ -77,18 +74,19 @@ def LoadModel(self, request, context):
         """
         model_name = request.Model
 
-        compute = "auto"
+        compute = torch.float16
         if request.F16Memory == True:
             compute=torch.bfloat16
 
-        self.CUDA = request.CUDA
+        self.CUDA = torch.cuda.is_available()
```
Comment (mudler) [inline backend/python/transformers/backend.py:81]: would make sense probably to keep reading `request.CUDA` ? maybe someone might want to turn it of explicitly afterall. Not a blocker

File: backend/python/transformers/requirements.txt (hunk 1 above)
Comment (fakezeta) [reply backend/python/transformers/requirements.txt:7]: the package belongs there but requirements-intel has `--extra-index-url https://pytorch-extension.intel.com/release-whl/stable/xpu/us/` and intel-extension-for-transformers is not in that repo. I had error during installation putting it there.

File: backend/python/transformers/backend.py (hunk 2 above)
Comment (fakezeta) [reply backend/python/transformers/backend.py:81]: It definitely make sense but request.CUDA by default is false, this means that it should be always set to true to use it. I think this way is more user friendly.
I can change if you want: np.

**Review by mudler — APPROVED** [review]

**Pull request merged**